iosApp/Survey/Sources/Presentation/Modules/SurveyDetail/SurveyQuestion/SurveyQuestionView.swift

```diff
@@ -6,28 +6,41 @@
 //  Copyright © 2023 Nimble. All rights reserved.
 //
 
+import Shared
 import SwiftUI
 
 struct SurveyQuestionView: View {
 
+    let detail: SurveyDetailUiModel
+
+    @Binding var questionIndex: Int
+
     var body: some View {
         VStack(alignment: .leading) {
-            // TODO: Use real data
-            Text("1/5")
+            Text("\(questionIndex + 1)/\(detail.questions.count)")
                 .font(.boldMedium)
                 .foregroundColor(.white)
                 .opacity(0.5)
                 .padding(.top, .largePadding)
                 .accessibility(.surveyQuestion(.detailText))
-            Text("How are you?")
+            Text(detail.questions[questionIndex].text)
                 .font(.boldLargeTitle)
                 .foregroundColor(.white)
                 .padding(.top, .tinyPadding)
                 .accessibility(.surveyQuestion(.titleText))
             Spacer()
-            QuestionPickerView(ids: ["A", "B", "C"])
+            questionView(with: detail.questions[questionIndex])
```

Comment:
This looks a bit dangerous if we accidentally have an invalid `questionIndex` somehow later on. Having an if check to show these views might be a better call, what do you think? 🤔

Reply:
Updated with tests.